modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -1568,6 +1571,28 @@ ABSDIFFC_SIMD(float)
 
 #undef ABSDIFFC_SIMD
 
+//-------------------------
+//
+// Fluid kernels: Split3
+//
+//-------------------------
+
+int split3_simd(const uchar in[], uchar out1[], uchar out2[],
+                uchar out3[], const int width)
+{
+    constexpr int nlanes = v_uint8::nlanes;
+    int x = 0;
+    v_uint8 a, b, c;
```

Comment:
> v_uint8 a, b, c;

Why is it outside of the loop body?

Keep variables declaration near their usage.

Reply:
Is it necessary to create vectors at each iteration?

Reply:
Is necessary to keep vector values between iterations?

Reply:
> Is necessary to keep vector values between iterations?
@alalek 
There is no necessity to do `setzero()` for each iteration of the loop since they are initialized by the `v_deinterleave()` function. So creation and initialization can be moved from the loop.

Reply:
Where do you see "setzero()" here?

Good rule of thumb for local variables to declare them in place where they are used.
No need to cross `for loop` boundary or any other, especially for SIMD register wrappers.

Reply:
> Where do you see "setzero()" here?

`_mmx_setzero_sixxx() `is called in the constructor of the `v_uint8` structure. In case of creation of 3 instances of the `v_uint8` strfucture the `_mmx_setzero_sixxx()` will be called 3 times. If we move creation of `v_uint8 a, b, c`  to the loop, `_mmx_setzero_sixxx()` will be called 3*width times. So this moving `v_uint8 a, b, c`  to the loop will affect performance.

Reply:
Done